ENIGMAsystem/SHELL/Platforms/General/PFshell/PFshell.cpp

```diff
@@ -0,0 +1,46 @@
+#include "Platforms/General/PFmain.h"
+#include "Platforms/General/PFshell.h"
+#include "Widget_Systems/widgets_mandatory.h"
+
+static bool secure_mode = true;
+static bool permission_requested = false;
+
+static void request_permission() {
+  if (!permission_requested) {
+    if (enigma_user::show_question("This application is requesting permission to use the command line. \
+This is a security risk and it is advised to only allow it from applications whose publishers \
+can be trusted.\n\nAllow the application to use such functionailty?")) {
+      secure_mode  = false; 
+    }
+      permission_requested = true;
+  }
+}
+
+namespace enigma_user {
+
+bool securemode_get_enabled() {
+  return secure_mode;
+}
+
+void securemode_set_enabled(bool enable) {
+  permission_requested = !enable;
+  secure_mode = enable;
+}
```

Comment:
The purpose of this secure mode is *exactly* so the application doesn't just get to say "your computer is mine and I will now call out to any external programs I want." I am fine with creating some sort of environment variable for this so that developers don't have to okay their own program every time they go to run their game, but this lets developers okay themselves when *I* go to run their game.

Reply:
@JoshDreamland can you *please* not be this incredibly stupid? fundies is using you as a pawn of my frustration. No one will even know what this function is, it's not even documented and only people who are smart enough to know what the source code reads as will know, and those are the users who are competent enough to be trusted with such a feature.